scripts/Muon/GUI/Common/fitting_widgets/basic_fitting/basic_fitting_model.py

```diff
@@ -823,13 +823,23 @@ def validate_sequential_fit(self, workspace_names: list) -> str:
         else:
             return ""
 
-    def get_runs_groups_and_pairs_for_fits(self) -> tuple:
+    def get_runs_groups_and_pairs_for_fits(self, display_type: str) -> tuple:
         """Returns the runs and group/pairs corresponding to the selected dataset names."""
         runs, groups_and_pairs = [], []
         for name in self.fitting_context.dataset_names:
             runs.append(get_run_numbers_as_string_from_workspace_name(name, self.context.data_context.instrument))
             groups_and_pairs.append(get_group_or_pair_from_name(name))
-        return self.fitting_context.dataset_names, runs, groups_and_pairs
+        return self._get_datasets_containing_string(display_type, self.fitting_context.dataset_names, runs,
+                                                    groups_and_pairs)
+
+    @staticmethod
+    def _get_datasets_containing_string(display_type: str, dataset_names: list, *corresponding_dataset_args) -> tuple:
+        """Returns the dataset names that contain a string and returns its associated runs/groups/pairs."""
+        if display_type == "All":
+            return (dataset_names, *corresponding_dataset_args)
+
+        # Filter the data based on a name in the dataset_names containing a string
+        return zip(*filter(lambda x: display_type in x[0], zip(dataset_names, *corresponding_dataset_args)))
```

Comment:
I find this a bit difficult to understand, could you explain it to me?

Reply:
It zips together the `dataset_names` and a couple of corresponding variables such as the run numbers in each of the dataset names. Then it uses `filter` to filter out any names that do not contain the `display_type` string.

It then returns a tuple containing the dataset names in one list, and the corresponding filtered run numbers (for example) in another list.

A similar thing is done in model fitting, except instead of filtering out data, it sorts the data https://github.com/mantidproject/mantid/blob/17357e04f8be38e419999bf09e30bf84cf9b424f/scripts/Muon/GUI/Common/fitting_widgets/model_fitting/model_fitting_model.py#L167

Reply:
so is x the return value from the inner zip?

Reply:
No, x is the parameter passed into the lambda (i.e. the inner zipped data). So when it does `x[0]` it is looking at the first item in the zipped item which is a dataset name. So `display_type in x[0]` checks if the `display_type` string is in the dataset name

Reply:
ah ok, that makes more sense.